posts/inside-rust/2024-06-19-this-development-cycle-in-cargo-1.80.md

```diff
@@ -0,0 +1,521 @@
+  - [`.crate` provenance](#crate-provenance)
+  - [`cargo publish --workspace`](#cargo-publish---workspace)
+  - [Snapshot testing](#snapshot-testing)
+- [Design discussions](#design-discussions)
+  - [RFC triage](#rfc-triage)
+  - [Custom test harnesses and `panic = "abort"`](#custom-test-harnesses-and-panic--abort)
+  - [Short-hand manifest syntaxes](#short-hand-manifest-syntaxes)
+  - [Leaky abstractions of rustc](#leaky-abstractions-of-rustc)
+- [Misc](#misc)
+- [Focus areas without progress](#focus-areas-without-progress)
+
+## Plugin of the cycle
+
+Cargo can't be everything to everyone,
+if for no other reason than the compatibility guarantees it must uphold.
+Plugins play an important part of the Cargo ecosystem and we want to celebrate them.
+
+Our plugin for this cycle is [cargo-expand](https://crates.io/crates/cargo-expand) a more convenient and easier to remember wrapper around `cargo rustc` for seeing all macros expanded within a crate.
+For macro authors, this is a big debugging help.
+For macro users, this can help in breaking the opacity of what a macro invocation does.
+
+Thanks to [LukeMathWalker](https://github.com/LukeMathWalker) for the suggestion!
+
+[Please submit your suggestions for the next post.](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Plugin.20of.20the.20Dev.20Cycle/near/420703211)
+
+## Implementation
+
+##### check-cfg
+
+*Update from [1.77](https://blog.rust-lang.org/inside-rust/2024/02/13/this-development-cycle-in-cargo-1-77.html#-zcheck-cfg)*
+
+As a refresher, this is a rustc feature that checks `#[cfg]`s against a list of known names and values.
+When used with Cargo, the names and values come from:
+- rustc's list of "well known" names (generally first party compilation toolchains)
+- cargo's list of "well known" names (generally first party crates.io ecosystem)
+- [`[features]`](https://doc.rust-lang.org/cargo/reference/features.html)
+- [`cargo::rustc-check-cfg` build.rs directive](https://doc.rust-lang.org/nightly/cargo/reference/build-scripts.html#rustc-check-cfg)
+- Passing `--check-cfg` through `RUSTFLAGS`
+
+At the beginning of May, Cargo's support for `--check-cfg` was stabilized ([#13571](https://github.com/rust-lang/cargo/pull/13571)).
+When stabilizing, the team weighed
+the [crater results](https://github.com/rust-lang/rust/issues/120701#issuecomment-1937010106)
+and feedback from the multiple rounds of [Call for Testing](https://github.com/rust-lang/rfcs/pull/3013#issuecomment-1936648479)
+(which included tweaking the layout for This Week in Rust to improve visibility).
+Hand-in-hand with stabilization,
+[urgau](https://github.com/urgau) published a
+[blog post](https://blog.rust-lang.org/2024/05/06/check-cfg.html)
+to help people first exposed to this by upgrading nightly.
+
+Soon after this hit nightly, [rust-lang/rust#124800](https://github.com/rust-lang/rust/issues/124800) (and related issues) were open.
+People's concerns included:
+- The false positive rate
```

Comment:
maybe we can expand a bit on the false positive rate; something like this maybe?
```suggestion
- The false positive rate (in particular, the sheer number of warnings that made it difficult/impossible to determine whether or not there was a problem)
```

Reply:
For myself, I don't feel this is needed and doesn't quite fit